src/lib.rs

```diff
@@ -224,8 +305,112 @@ mod tests {
     use super::*;
 
     #[test]
-    fn seq() {
+    fn seq ()
+    {
         let seq: [usize; 5] = array_init(|i| i);
         assert_eq!(&[0, 1, 2, 3, 4], &seq);
     }
+
+    #[test]
+    fn array_from_iter ()
+    {
+        let array = [0, 1, 2, 3, 4];
+        let seq: [usize; 5] = from_iter(array.iter().copied()).unwrap();
+        assert_eq!(
+            array,
+            seq,
+        );
+    }
+
+    #[test]
+    fn array_init_no_drop ()
+    {
+        DropChecker::with(|drop_checker| {
+            let result: Result<[_; 5], ()> =
+                try_array_init(|i| {
+                    if i < 3 {
+                        Ok(drop_checker.new_element())
+                    } else {
+                        Err(())
+                    }
+                })
+            ;
+            let _ = result.map(|_array| unreachable!());
```

Comment:
You should assert on `Result::is_err` for clarity

Reply:
Done.